src/GroupBox.js

```diff
@@ -0,0 +1,125 @@
+    const keys = Object.keys(currentParentObj)
+    keys.map(key => {
+      if (currentParentObj[key].checked) {
+        ticked++
+      }
+    })
+    if (!ticked || ticked < minimumRequired) {
+      this.setState({error: true})
+      keys.forEach(key => {
+        formObj[key].error = true
+        formObj[key].errorMsg = errorMsg
+      })
+    } else {
+      keys.forEach(key => {
+        formObj[key].error = false
+        formObj[key].errorMsg = ''
+      })
+      this.setState({error: false})
+    }
+    return formObj
+  }
+  getParentId() {
+    return ++parentId
+  }
+  componentWillMount() {
+    this.parentId = `group${this.getParentId()}`
+  }
+  componentWillUnMount() {
+    parentId = 0
+  }
+  getChildrenWithProps(props, parentElement) {
+    const {children} = props
+    let updateChild
+    let childElement
+    if (Array.isArray(children)) {
+      const updateChildren = children.map(item => {
+        const type = this.getType(item.props)
+        if (type === 'checkbox' || type === 'radio') {
+          updateChild = React.cloneElement(item, {onSubmitCallBack: this.onSubmitCallBack, parentId: this.parentId})
+        } else if (item.props.children) {
+          childElement = this.getChildrenWithProps(item.props)
+          updateChild = React.cloneElement(item, {children: childElement})
+        }
+        return updateChild
+      })
+      return updateChildren
+    } else if (children) {
+      const childType = this.getType(children.props)
+      let subLevelChild
+      if (childType === 'checkbox' || childType === 'radio') {
+        return React.cloneElement(children, {onSubmitCallBack: this.onSubmitCallBack, parentId: this.parentId})
+      } else if (children.props && children.props.children) {
+        subLevelChild = this.getChildrenWithProps(children.props)
+        return React.cloneElement(children, {children: subLevelChild})
+      } else {
+        return children
+      }
+    }
+  }
+  getType(props) {
+    const inputTypeJson = props && props.inputTypeJson
+    const inputProps = inputTypeJson && inputTypeJson.inputProps
+    const type = inputProps && inputProps.type
+    return type
+  }
+  displayError() {
+    const {inputTypeJson} = this.props
+    const {errorMsg, ErrorComponent} = inputTypeJson && inputTypeJson.errorProps
+
+    if (ErrorComponent) {
+      return <ErrorComponent />
+    }
+    return (
+      <DefaultErrorComponent >{errorMsg}</DefaultErrorComponent>
+    )
+  }
+  render() {
+    const {error} = this.state
+    const {inputTypeJson} = this.props
+    let className
+    let id
+    let errorProps
+    if (inputTypeJson) {
+      errorProps = inputTypeJson.errorProps
+      className = inputTypeJson.className
+      id = inputTypeJson.id
+    }
+    const {renderErrorAfter, renderErrorBefore} = errorProps || {}
+    const updateChildren = this.getChildrenWithProps(this.props)
+    return (
+      <div className={className} id={id}>
```

Comment:
Discussion needed over here
@Ayush-krishnatray  do we need to put this div for grouping checkbox/radiobox I have put this div so that in case of error we can give some border on it but it will change the structure of dom and can cause some problem in user grid.
if we do not keep it then if user need to highlight the groupbox then he need to write some piece of code themselves.

Reply:
no we don't user can put his own wrapper, which will also reduce the number of props required.

Reply:
done